### PR TITLE
Lu 4635

### PR DESCRIPTION
**Pull request by DarKim59**
Changes made to decrypt plain text password where necessary

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,6 +8,12 @@ Clone the git respository to your IDE of choice. Visual Studio 2019 is recomende
 
 Populate the key values in the App.config file accordingly. **Never committ App.config with key values.**
 
+Please note that there is a 'token' config setting in the app config - this needs to be set to an appropriate alphanumeric string, as per the documentation here https://cryptography.io/en/latest/fernet/  In terms of Blaise the token is held against the Azure release pipeline variable and replaced during the release of this C# service.
```
Comment (salmaniqbal) [inline README.md]: What should it be set to for each environment?

File: BlaiseUserSync/BlaiseUserSync.cs [hunk 2]
```diff
@@ -264,20 +263,81 @@ public static IConnectedServer2 ConnectToBlaiseServer()
...
+        /// <returns>plain text string</returns>
+        private static bool DecryptString(string encryptedString, out string decryptedString)
+        {
+            decryptedString = string.Empty;
+
+            try
+            {
+                DateTime tsNow = DateTime.Now;
+                // Get the token from the app config
+                log.Info("Getting pw token: " + ConfigurationManager.AppSettings["Token"]);
```
Comment (salmaniqbal) [inline BlaiseUserSync/BlaiseUserSync.cs]: Not sure if printing the token is necessary.

File: BlaiseUserSync/BlaiseUserSync.cs [hunk 3]
```diff
@@ -264,20 +263,81 @@ public static IConnectedServer2 ConnectToBlaiseServer()
...
+
+                char[] passwordChars = pw.ToCharArray();
+                foreach (char c in passwordChars)
+                {
+                    password.AppendChar(c);
+                }
+            }
+            catch (ApplicationException ae)
+            {
+                log.Error("Error in GetPassword - DecryptString() - Application Exception Raised: " + ae.Message);
```
Comment (salmaniqbal) [inline BlaiseUserSync/BlaiseUserSync.cs]: just check that exception does not log the password

Comment (DarKim59) [reply BlaiseUserSync/BlaiseUserSync.cs]: No, its doesn't:- 2020-01-15 13:56:55,830 [1] INFO BlaiseUserSync.BlaiseUserSync - Queue declared - ONS24158 
2020-01-15 13:56:55,836 [1] INFO BlaiseUserSync.BlaiseUserSync - Queue binding complete - ONS24158 / BSM_Exchange / ONS24158 
2020-01-15 13:56:55,850 [1] INFO BlaiseUserSync.BlaiseUserSync - Consumer object created. 
2020-01-15 13:56:55,850 [1] INFO BlaiseUserSync.BlaiseUserSync - RabbitMQ setup complete. 
2020-01-15 13:56:59,202 [WorkPool-Session#1:Connection(5b176987-bd91-4e59-9f74-ff8d4fdc4569,amqp://localhost:5672)] INFO BlaiseUserSync.BlaiseUserSync - Message received: {"ACTION": "CHANGE_PASSWORD", "username": "Darren2", "password": "gAAAAABeHa-WBoSQzVyjvppuALHdBLqX70ZZzmkxhIgiFtlkb5_LI2eBP5YFrESqry2xXgPlbKOMCjb60EA-HGR7AI4EuTxJvA=="} 
2020-01-15 13:58:09,598 [WorkPool-Session#1:Connection(5b176987-bd91-4e59-9f74-ff8d4fdc4569,amqp://localhost:5672)] ERROR BlaiseUserSync.BlaiseUserSync - Error in GetPassword - DecryptString() - Application Exception Raised: Error in the application.

File: BlaiseUserSync/BlaiseUserSync.cs (hunk 2 above)
Comment (DarKim59) [reply BlaiseUserSync/BlaiseUserSync.cs]: yes agree - have removed this now

File: README.md (hunk 1 above)
Comment (DarKim59) [reply README.md]: this is going to change as I think we are going to hold the token (or key) in Kubernetes secrets as of 16/01/2020

Comment (DarKim59) [reply README.md]: yes we should have different tokens (keys) for different envs e.g.: 

For TEST / DEV: fTEsTtEst28TeStTeStTES2tESTTeSTestsTestsTtT=
For UAT / Pre-prod:  [a random generated one]
For LIVE: [a random generated one - so different to UAT / pre-prod]]

**Pull request merged**